nemo/collections/tts/fastspeech/data.py

```diff
@@ -0,0 +1,186 @@
+        self._durs_dir = durs_dir
+
+    def __getitem__(self, index):
+        audio, audio_len, text, text_len = self._audio_dataset[index]
+        dur_true = torch.tensor(np.load(os.path.join(self._durs_dir, f'{index}.npy'))).long()
+        return dict(audio=audio, audio_len=audio_len, text=text, text_len=text_len, dur_true=dur_true)
+
+    def __len__(self):
+        return len(self._audio_dataset)
+
+
+class FastSpeechDataLayer(DataLayerNM):
+    """Data Layer for Fast Speech model.
+
+    Basically, replicated behavior from AudioToText Data Layer, zipped with ground truth durations for additional loss.
+
+    Args:
+        manifest_filepath (str): Dataset parameter.
+            Path to JSON containing data.
+        durs_dir (str): Path to durations arrays directory.
+        labels (list): Dataset parameter.
+            List of characters that can be output by the ASR model.
+            For Jasper, this is the 28 character set {a-z '}. The CTC blank
+            symbol is automatically added later for models using ctc.
+        batch_size (int): batch size
+        sample_rate (int): Target sampling rate for data. Audio files will be
+            resampled to sample_rate if it is not already.
+            Defaults to 16000.
+        int_values (bool): Bool indicating whether the audio file is saved as
+            int data or float data.
+            Defaults to False.
+        eos_id (id): Dataset parameter.
+            End of string symbol id used for seq2seq models.
+            Defaults to None.
+        min_duration (float): Dataset parameter.
+            All training files which have a duration less than min_duration
+            are dropped. Note: Duration is read from the manifest JSON.
+            Defaults to 0.1.
+        max_duration (float): Dataset parameter.
+            All training files which have a duration more than max_duration
+            are dropped. Note: Duration is read from the manifest JSON.
+            Defaults to None.
+        normalize_transcripts (bool): Dataset parameter.
+            Whether to use automatic text cleaning.
+            It is highly recommended to manually clean text for best results.
+            Defaults to True.
+        trim_silence (bool): Whether to use trim silence from beginning and end
+            of audio signal using librosa.effects.trim().
+            Defaults to False.
+        load_audio (bool): Dataset parameter.
+            Controls whether the dataloader loads the audio signal and
+            transcript or just the transcript.
+            Defaults to True.
+        drop_last (bool): See PyTorch DataLoader.
+            Defaults to False.
+        shuffle (bool): See PyTorch DataLoader.
+            Defaults to True.
+        num_workers (int): See PyTorch DataLoader.
+            Defaults to 0.
+        perturb_config (dict): Currently disabled.
+
+    """
+
+    @property
+    def output_ports(self) -> Optional[Dict[str, NeuralType]]:
```

Comment:
when you are defining new neural modules, please put "add_port_docs" decorator like here https://github.com/NVIDIA/NeMo/blob/master/nemo/collections/tts/tacotron2_modules.py#L37 so that nicer autodocs would be generated

Reply:
the same comment is probably true for all your other input/output ports